src/carts/src/main/java/mushop/carts/CartService.java

```diff
@@ -14,159 +17,192 @@
     }
-    
+
     /**
      * POST /{cartId}
      * 
-     * Adds or updates a cart.  If the cartId does not exist, a new cart is added.  
-     * If the cartId does exist, the given cart is merged in to the existing one.
-     * This method is used to add additional items to a cart. 
+     * Adds or updates a cart. If the cartId does not exist, a new cart is added. If
+     * the cartId does exist, the given cart is merged in to the existing one. This
+     * method is used to add additional items to a cart.
      */
     public void postCart(ServerRequest request, ServerResponse response) {
         String cartId = request.path().param("cartId");
-        
+
         try {
             request.content().as(Cart.class).thenAccept(newCart -> {
-                Cart cart = carts.getById(cartId);
-                if (cart == null) {
-                    
-                    newCart.setId(cartId);
-                    carts.save(newCart);
-                    response.status(201).send(); // created
-                } else {
-                    cart.merge(newCart);
-                    carts.save(cart);
-                    response.status(200).send(); // ok
+                try {
+                    Cart cart = carts.get().getById(cartId);
+                    if (cart == null) {
+
+                        newCart.setId(cartId);
+                        carts.get().save(newCart);
+                        response.status(201).send(); // created
+                    } else {
+                        cart.merge(newCart);
+                        carts.get().save(cart);
+                        response.status(200).send(); // ok
+                    }
+                } catch (Exception e) {
+                    sendError(response, e.getMessage());
+                    return;
                 }
             });
         } catch (Exception e) {
             sendError(response, e.getMessage());
             return;
         }
     }
-    
+
     /**
      * PUT /{cartId}/items
      * 
-     * Updates the quantity of an item in the cart. 
+     * Updates the quantity of an item in the cart.
      */
     public void updateCartItem(ServerRequest request, ServerResponse response) {
         String cartId = request.path().param("cartId");
         try {
             request.content().as(Item.class).thenAccept(qItem -> {
-                Cart cart = carts.getById(cartId);
-                if (cart == null) {
+                try {
+                    Cart cart = carts.get().getById(cartId);
+                    if (cart == null) {
+                        response.status(404).send();
+                        return;
+                    }
+                    for (Item item : cart.getItems()) {
+                        if (item.getItemId().equals(qItem.getItemId())) {
+                            item.setQuantity(qItem.getQuantity());
+                            carts.get().save(cart);
+                            response.status(200).send();
```

Comment:
you can probably return here after the `.send()` so it's not iterating through the cart items anymore.

Reply:
Yeah - I noticed this too - I think the 
```
carts.get().save(cart);
response.status(200).send();
```
needs to be outside the loop.

this is the original logic here, and I will fix this in a separate PR